packages/perseus/src/widgets/dropdown/dropdown.tsx

```diff
@@ -29,143 +34,182 @@ type Props = WidgetProps<
     dependencies: PerseusDependenciesV2;
 };
 
-type DefaultProps = {
-    choices: Props["choices"];
-    placeholder: Props["placeholder"];
-    apiOptions: Props["apiOptions"];
-    userInput: Props["userInput"];
+// Widget interface methods exposed via ref
+type WidgetHandle = {
+    focus: () => boolean;
+    getPromptJSON: () => DropdownPromptJSON;
+    getDOMNodeForPath: (path: FocusPath) => Element | null;
+    getSerializedState: () => any;
 };
 
-class Dropdown extends React.Component<Props> implements Widget {
-    static contextType = PerseusI18nContext;
-    declare context: React.ContextType<typeof PerseusI18nContext>;
-
-    static defaultProps: DefaultProps = {
-        choices: [],
-        placeholder: "",
-        apiOptions: ApiOptions.defaults,
-        userInput: {value: 0},
-    };
-
-    componentDidMount(): void {
-        this.props.dependencies.analytics.onAnalyticsEvent({
+const Dropdown = forwardRef<WidgetHandle, Props>((props, ref) => {
```

Comment:
I believe it's a little bit better to use `function` instead of `const` for React functional components. I started a thread in the web frontend Slack if you'd like to see the conversation around it!